src/github.com/getlantern/flashlight/client/client.go

```diff
@@ -179,3 +191,29 @@ func (client *Client) Configure(cfg *ClientConfig, proxyAll func() bool) {
 func (client *Client) Stop() error {
 	return client.l.Close()
 }
+
+func (client *Client) proxiedDialer(orig func(network, addr string) (net.Conn, error)) func(network, addr string) (net.Conn, error) {
+	var proxied func(network, addr string) (net.Conn, error)
+	if client.ProxyAll() {
+		proxied = orig
+	} else {
+		proxied = detour.Dialer(orig)
+	}
+
+	return func(network, addr string) (net.Conn, error) {
+		if isLanternSpecialDomain(addr) {
```

Comment:
This is the magic.  The UI is actually accessed via the LanternSpecialDomain, which we detect here and then dial to the actual UI server.